modules/programs/radio-active.nix

```
@@ -0,0 +1,188 @@
+              filetype = mp3
+              player = ffplay
+              ```
+            '';
+
+            type = attrsOf (oneOf [
+              nonEmptyStr
+              nonnegative
+            ]);
+          };
+
+          aliases = mkOption {
+            description = ''
+              Key/value pairs where the key is name of radio station and value is URL
+            '';
+
+            example = ''
+              ## Input
+
+              ```nix
+              programs.radio-active.aliases = {
+                "Deep House Lounge" = "http://198.15.94.34:8006/stream";
+              };
+              ```
+
+              ## Output
+
+              ```
+              Deep House Lounge==http://198.15.94.34:8006/stream
+              ```
+            '';
+
+            default = { };
+            type = attrsOf nonEmptyStr;
+          };
+        };
+      };
+    };
+  };
+
+  config = mkIf cfg.enable {
+    ## TODO: test that dependency `postPatch` modifications works at runtime
+    home.packages =
+      let
+        player = attrByPath [ "settings" "AppConfig" "player" ] "ffplay" cfg;
+
+        radio-active =
+          if player == "ffplay" then
+            pkgs.radio-active
+          else if player == "mpv" then
+            pkgs.radio-active.overrideAttrs (
+              finalAttrs: previousAttrs: {
+                postPatch = ''
+                  ${previousAttrs.postPatch}
+
+                  substituteInPlace radioactive/mpv.py \
+                    --replace-fail 'self.exe_path = which(self.program_name)' \
+                    'self.exe_path = "${lib.getExe pkgs.mpv}"'
+                '';
+              }
+            )
+          else if player == "vlc" then
+            pkgs.radio-active.overrideAttrs (
+              finalAttrs: previousAttrs: {
+                postPatch = ''
+                  ${previousAttrs.postPatch}
+
+                  substituteInPlace radioactive/vlc.py \
+                    --replace-fail 'self.exe_path = which(self.program_name)' \
+                    'self.exe_path = "${lib.getExe pkgs.vlc}"'
+                '';
+              }
+            )
+          else
+            lib.warn ''
+              Unknown player defined in `config.programs.radio-active.AppConfig.player`
+            '' pkgs.radio-active;
+      in
+      mkIf (cfg.package != null) [
+        radio-active
+      ];
+
+    xdg.configFile."radio-active/configs.ini" =
+      lib.mkIf (cfg.settings != { } && cfg.settings.AppConfig != { })
+        {
+          source = iniFormat.generate "radio-active-config" {
+            inherit (cfg.settings) AppConfig;
+          };
+        };
+
+    home.file."radio-active-alias" = mkIf (cfg.settings != { } && cfg.settings.aliases != { }) {
+      text = ''
+        ${builtins.concatStringsSep "\n" (
+          mapAttrsToList (name: value: "${name}==${value}") cfg.settings.aliases
+        )}
+      '';
+      target = "${config.home.homeDirectory}/.radio-active-alias";
```

No need for the `target` attribute:

```suggestion
    home.file.".radio-active-alias" = mkIf (cfg.settings != { } && cfg.settings.aliases != { }) {
      text = ''
        ${builtins.concatStringsSep "\n" (
          mapAttrsToList (name: value: "${name}==${value}") cfg.settings.aliases
        )}
      '';
```